faf-java-server-app/src/main/java/com/faforever/server/game/GameService.java

```diff
@@ -836,38 +895,29 @@ protected void onGameEnded(Game game) {
   }
 
   /**
-   * Returns the active game with the specified ID, if such a game exists. A game is considered active as soon as it's
-   * being hosted and until it finished.
+   * Takes a map of armyId->ArmyResult and maps all army ids to player ids.
    */
   @VisibleForTesting
-  Optional<Game> getActiveGame(int id) {
-    return activeGameRepository.findById(id);
-  }
-
-  /**
-   * Determines the final army results and updates the players' scores with them.
-   */
-  private void settlePlayerScores(Game game) {
-    Map<Integer, ArmyResult> armyIdToResult = findMostReportedCompleteArmyResultsReportedByConnectedPlayers(game);
-
+  Map<Integer, ArmyResult> mapArmyResultsToPlayerIds(Game game, Map<Integer, ArmyResult> armyIdToResult) {
+    Map<Integer, ArmyResult> playerIdToResult = new HashMap<>();
     for (GamePlayerStats gamePlayerStats : game.getPlayerStats().values()) {
-      Integer playerId = gamePlayerStats.getPlayer().getId();
-      Integer armyScore = Optional.ofNullable(game.getPlayerOptions())
+      final Integer playerId = gamePlayerStats.getPlayer().getId();
```

Comment:
It's ok to do `final` but we usually don't, and since Java 8 it's optional AFAIK